public/app/features/commandPalette/CommandPalette.tsx

```diff
@@ -80,24 +80,34 @@ const RenderResults = ({ dashboardResults }: RenderResultsProps) => {
     () => dashboardResults.map((dashboard) => new ActionImpl(dashboard, { store: {} })),
     [dashboardResults]
   );
+
   const items = useMemo(
     () => (dashboardResultItems.length > 0 ? [...results, dashboardsSectionTitle, ...dashboardResultItems] : results),
     [results, dashboardsSectionTitle, dashboardResultItems]
   );
 
   return (
-    <div className={styles.resultsContainer}>
-      <KBarResults
-        items={items}
-        onRender={({ item, active }) =>
+    <KBarResults
+      items={items}
+      onRender={({ item, active }) => {
+        // These items are rendered in a container, in a virtual list, so we cannot
+        // use :first/last-child selectors, so we must mimic them in JS
+        const isFirstItem = items[0] === item;
+        const isLastItem = items[items.length - 1] === item;
+
+        const renderedItem =
           typeof item === 'string' ? (
-            <div className={styles.sectionHeader}>{item}</div>
+            <div className={styles.sectionHeader}>
+              <div className={cx(styles.sectionHeaderInner, isFirstItem && styles.sectionHeaderInnerFirst)}>{item}</div>
+            </div>
```

Comment:
Not in love with how i did this, but totally open to better suggestions!

I tried using a distinct `<hr />`-ish element for the border, but it ended up just as fiddly to get the padding right.

Reply:
The only alternative I can find is to play around with margin and padding... This css achieves the same with only a `sectionHeader` (no `sectionHeaderInner`), but I think it might be _more_ hacky:
```
  sectionHeader: css({
    fontSize: theme.typography.h6.fontSize,
    fontWeight: theme.typography.body.fontWeight,
    color: theme.colors.text.secondary,
    padding: theme.spacing(1, 2, 3),
    borderTop: `1px solid ${theme.colors.border.medium}`,
    marginTop: theme.spacing(2),
  }),
```

Reply:
When I used margin the elements would be positioned in the virtualised list, and so it would look all wonky…

Reply:
Yeah I had to very hackily add bottom padding to make up for the margin not counting for the height of the item 👀  Ill approve this PR, I think your solution is the best one